koku/masu/database/report_manifest_db_accessor.py

```diff
@@ -111,3 +117,30 @@ def get_manifest_list_for_provider_and_bill_date(self, provider_uuid, bill_date)
         """Return all manifests for a provider and bill date."""
         filters = {"provider_id": provider_uuid, "billing_period_start_datetime__date": bill_date}
         return CostUsageReportManifest.objects.filter(**filters).all()
+
+    def get_last_seen_manifest_ids(self, bill_date):
+        """Return a tuple containing the assembly_id of the last seen manifest and a boolean
+
+        The boolean will state whether or not that manifest has been processed."""
+        assembly_ids = []
+        # The following query uses a window function to rank the manifests for all the providers,
+        # and then just pulls out the top ranked (most recent) manifests
+        manifests = (
+            CostUsageReportManifest.objects.filter(billing_period_start_datetime=bill_date)
+            .annotate(
+                row_number=Window(
+                    expression=RowNumber(),
+                    partition_by=F("provider_id"),
+                    order_by=F("manifest_creation_datetime").desc(),
+                )
+            )
+            .order_by("row_number")
+        )
+        for manifest in manifests:
```

Comment:
There's one more thing we need to check here. I fiddled with the ORM query to try to get this done in SQL, but it's annoyingly difficult to figure out, so in this loop you want to only get the manifests where `row_number == 1`. When row number is `1` that means it is the most recent manifest.

Reply:
AHHH! That actually makes a ton of sense. Thank you for catching that!